recipes/CodeLLM_DevKit/generate_unit_tests.ipynb

```diff
@@ -0,0 +1,373 @@
+    "\n",
+    "        # Generate test cases for non-interface and non-abstract classes\n",
+    "        if not class_details.is_interface and \"abstract\" not in class_details.modifiers:\n",
+    "            \n",
+    "            # Get all constructor signatures\n",
+    "            constructor_signatures = \"\"\n",
+    "            \n",
+    "            for method in analysis.get_methods_in_class(qualified_class_name=class_name):\n",
+    "                method_details = analysis.get_method(qualified_class_name=class_name, qualified_method_name=method)\n",
+    "                \n",
+    "                if method_details.is_constructor:\n",
+    "                    constructor_signatures += method_details.signature + '\\n'\n",
+    "            \n",
+    "            # If no constructor present, then add the signature of the default constructor\n",
+    "            if constructor_signatures == \"\":\n",
+    "                constructor_signatures = f\"public {focal_class_name}() \" + \"{}\"\n",
+    "            \n",
+    "            # Go through all the methods in the class\n",
+    "            for method in analysis.get_methods_in_class(qualified_class_name=class_name):\n",
+    "                \n",
+    "                if method == focal_method:\n",
+    "                    # Get the method details\n",
+    "                    method_details = analysis.get_method(qualified_class_name=class_name, qualified_method_name=method)\n",
+    "                    \n",
+    "                    # Generate test cases for non-private methods\n",
+    "                    if \"private\" not in method_details.modifiers and not method_details.is_constructor:\n",
+    "                        \n",
+    "                        # Gather all the information needed for the prompt, which are focal method body, focal method name, focal class name, and constructor signature\n",
+    "                        prompt = format_inst(\n",
+    "                            focal_method_body=method_details.declaration+method_details.code,\n",
+    "                            focal_method=method.split(\"(\")[0],\n",
+    "                            focal_class=focal_class_name,\n",
+    "                            constructor_signatures=constructor_signatures,\n",
+    "                            language=\"java\"\n",
+    "                        )\n",
+    "                        \n",
+    "                        # Print the instruction\n",
+    "                        print(f\"Instruction:\\n{prompt}\\n\")\n",
+    "                        print(f\"Generating test case ...\\n\")\n",
+    "                        \n",
+    "                        # Prompt the local model on Ollama\n",
+    "                        llm_output = prompt_ollama(message=prompt)\n",
+    "                        \n",
+    "                        # Print the LLM output\n",
+    "                        print(f\"LLM Output:\\n{llm_output}\")"
+   ]
+  }
+ ],
```

Comment:
The last cell just prints `analysis path: .`. 

Could you add a final summary markdown cell that explains what people should have seen?

Reply:
The cell should throw an exception if not.

Reply:
Added maven as a prerequisite

Reply:
@fayvor `Generating code summary and it will take few minutes (or even seconds) based on where the model has been hosted...`

Is it a good description. I can add that and create a PR real quick.